<a href="https://colab.research.google.com/github/angelcortez/titanic/blob/devAngelCortez/01_El_Undimiento_del_Titanic.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# El Hundimiento del Titanic

La noche del 14 de abril de 1912, el RMS Titanic colisionó con un iceberg en el océano Atlántico durante su viaje inaugural de Southampton a Nueva York. Considerado uno de los barcos más lujosos y avanzados de su época, el Titanic se hundió durante la madrugada del 15 de abril, convirtiéndose en uno de los desastres marítimos más conocidos de la historia. Sin embargo, detrás de este acontecimiento histórico existen patrones y fenómenos que pueden analizarse mediante herramientas probabilísticas y estadísticas.

En esta actividad utilizaremos el dataset del hundimiento del Titanic para investigar, desde una perspectiva matemática, cómo factores como el sexo, la clase social, la edad o el puerto de embarque pudieron influir en las probabilidades de supervivencia de los pasajeros. A través de este análisis se aplicarán conceptos fundamentales de teoría de probabilidad y procesos estocásticos, tales como probabilidades marginales, probabilidades condicionales, operaciones entre eventos, independencia y la Regla de Bayes. El objetivo es interpretar los datos como un experimento aleatorio real y utilizar el razonamiento probabilístico para extraer conclusiones sobre uno de los eventos históricos más emblemáticos del siglo XX.

---

## Definición de eventos

Considere los siguientes eventos:

* N: “el pasajero sobrevivió”
* M: “el pasajero es hombre”
* W: “el pasajero es mujer”
* F: “el pasajero viajó en primera clase”
* G: “el pasajero viajó en segunda clase”
* T: “el pasajero viajó en tercera clase”

Para cada evento expresa la probabilidad en notación matemática en el siguiente bloque de Markdown (recuerda que Markdown es compatible con código de LaTeX para incluir expresiones matemáticas):

$$ P(N) = \frac{\text{sobrevivientes}}{\text{total de pasajeros}} $$

$$ P(M) = \frac{\text{pasajeros hombres}}{\text{total de pasajeros}} $$

$$ P(W) = \frac{\text{pasajeras mujeres}}{\text{total de pasajeros}} $$

$$ P(F) = \frac{\text{pasajeros en primera clase}}{\text{total de pasajeros}} $$

$$ P(G) = \frac{\text{pasajeros en segunda clase}}{\text{total de pasajeros}} $$

$$ P(T) = \frac{\text{pasajeros en tercera clase}}{\text{total de pasajeros}} $$

In [3]:
# CARGA DEL DATASET

import pandas as pd
df = pd.read_csv('titanic_sink_data.csv')
print(df.head())

   PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S  


---
### Usando el dataset proporcionado realiza los cálculos pertinentes y responde a las siguientes preguntas

1. (Probabilidad marginal) ¿Cuál es la probabilidad de que un pasajero seleccionado al azar haya sobrevivido $P(N)$?

In [20]:
casos_favorables = len(df[df['Survived'] == 1])  # sobrevivientes
casos_totales = len(df)                          # total de pasajeros

pm_N = casos_favorables / casos_totales

print("P(N) = casos Favorables / casos totales")
print("P(N) = Pasajeros que sobrevivieron / Total de pasajeros en el barco")
print("-" * 65)
print(f"P(N) = {casos_favorables} / {casos_totales}")
print("-" * 65)
print(f"Probabilidad Marginal: P(N) = {pm_N:.4f} (o {pm_N * 100:.2f}%)")

P(N) = casos Favorables / casos totales
P(N) = Pasajeros que sobrevivieron / Total de pasajeros en el barco
-----------------------------------------------------------------
P(N) = 342 / 891
-----------------------------------------------------------------
Probabilidad Marginal: P(N) = 0.3838 (o 38.38%)


Respuesta: $P(N) = 38.38\% $

2. (Probabilidad conjunta) ¿Cuál es la probabilidad de que un pasajero sea mujer y además haya sobrevivido $P(W \cap N) $?

In [21]:
# '&' es el operador AND que indica que se deben cumplir las dos condiciones
mujeres_sobrevivientes = len(df[(df['Sex'] == 'female') & (df['Survived'] == 1)]) # mujeres sobrevivientes
casos_totales = len(df)                          # total de pasajeros

pc_WANDN = mujeres_sobrevivientes / casos_totales

print("P(W ∩ N) = casos favorables / casos totales")
print("P(W ∩ N) = (mujeres que sobrevivieron) / (total de pasajeros)")
print("-" * 65)
print(f"P(W ∩ N) = {mujeres_sobrevivientes} / {casos_totales}")
print("-" * 65)
print(f"Probabilidad conjunta: P(W ∩ N) = {pc_WANDN:.4f} (o {pc_WANDN * 100:.2f}%)")

P(W ∩ N) = casos favorables / casos totales
P(W ∩ N) = (mujeres que sobrevivieron) / (total de pasajeros)
-----------------------------------------------------------------
P(W ∩ N) = 233 / 891
-----------------------------------------------------------------
Probabilidad conjunta: P(W ∩ N) = 0.2615 (o 26.15%)


Respuesta: $P(W \cap N) = 26.15\%$

3. (Unión de eventos) ¿Cuál es la probabilidad de que un pasajero haya sobrevivido o viajado en primera clase? Toma en cuenta lo siguiente:

$$ P(N \cup F) = P(N) + P(F) - P(N\cap F) $$

In [6]:
# TODO

Respuesta: $P(N \cup F) =$

4. (Probabilidad condicional) ¿Cuál es la probabilidad de sobrevivir dado que el pasajero era mujer $P(N \mid W)$?

In [7]:
# TODO

Respuesta: $P(N \mid W) =$

5. (Probabilidad condicional inversa) ¿Cuál es la probabilidad de que un pasajero fuera mujer dado que sobrevivió $P(W \mid N)$?

In [8]:
# TODO

Respuesta: $P(W \mid N) =$

6. (Regla de Bayes) Utilizando la Regla de Bayes, calcule la probabilidad de que un pasajero haya viajado en primera clase dado que sobrevivió $P(F \mid N)$. Recuerda la *Regla de Bayes*:

$$P(A\mid B)=\frac{P(B\mid A)P(A)}{P(B)}$$

In [9]:
# TODO

Respuesta: $P(F \mid N) = ?$

7. (Independencia de eventos) Determina si los eventos:

* “ser hombre”
* “sobrevivir”

son independientes.

Verifique si:


$$P(M \cap N)=P(M)P(N)$$


e interprete el resultado.

In [10]:
# TODO

Respuesta:

8. (Complemento de un evento) ¿Cuál es la probabilidad de que un pasajero **NO** haya sobrevivido $P(N^{c})$?

In [11]:
# TODO

Respuesta: $P(N^{c}) = ?$

9. (Probabilidad condicional múltiple) ¿Cuál es la probabilidad de sobrevivir dado que el pasajero era mujer y viajaba en primera clase $P(N \mid W \cap F)$?

In [12]:
# TODO

Respuesta: $P(N \mid W \cap F) = ?$


10. (Comparación entre probabilidades condicionales) Compara las siguientes probabilidades:

$$ P(N \mid F), \quad P(N \mid G), \quad P(N \mid T) $$

y responde:

* ¿Qué clase tuvo mayor probabilidad de supervivencia?
* ¿Existe evidencia de desigualdad socioeconómica en la supervivencia?

evidencia tus respuestas gráficando la distribución de probabilidades.

In [13]:
# TODO